packages/components/src/tools-panel/tools-panel/hook.js

```diff
@@ -101,6 +102,37 @@ export function useToolsPanel( props ) {
 		setMenuItems( items );
 	}, [ panelItems ] );
 
+	// Track whether all optional controls are displayed or not.
+	// If no optional controls are present, then none are hidden and this will
+	// be `false`.
+	const [
+		areAllOptionalControlsHidden,
+		setAreAllOptionalControlsHidden,
+	] = useState( false );
+
+	// Where no optional menu items are active, we display a plus icon
+	// to indicate the presence of further menu items.
+	useEffect( () => {
+		if ( menuItems.optional ) {
+			const optionalItems = Object.entries( menuItems.optional );
+			const allControlsHidden =
+				optionalItems.length > 0 &&
+				! optionalItems.some( ( [ , isSelected ] ) => isSelected );
+			setAreAllOptionalControlsHidden( allControlsHidden );
+		}
+	}, [ menuItems.optional ] );
+
+	const hasDefaultMenuItems =
+		menuItems?.default && !! Object.keys( menuItems?.default ).length;
+
+	const emptyClassName =
+		! hasDefaultMenuItems && areAllOptionalControlsHidden && 'is-empty';
+
+	const cx = useCx();
+	const classes = useMemo( () => {
+		return cx( styles.ToolsPanel, emptyClassName, className );
+	}, [ className, emptyClassName ] );
```

Comment:
Although this part of the code may be removed from `ToolsPanel` (and moved to the `packages/block-editor/src/components/inspector-controls/` folder), I still wanted to give some feedback about not using hardcoded classnames.

In this case, instead of using `is-empty`, we could create a style object in `styles.ts` (let's say `toolsPanelEmpty` for example), and then import it like so:

```suggestion
	const classes = useMemo(
		() =>
			cx(
				styles.ToolsPanel,
				! hasDefaultMenuItems && areAllOptionalControlsHidden && styles.toolsPanelEmpty,
				className
			),
		[ className, hasDefaultMenuItems, areAllOptionalControlsHidden ]
	);
```

Reply:
The application of any styles indicating a blank display state for the panel will rely on checking the menu items state so can't be moved out of the ToolsPanel for the time being.

I've adopted your suggestion here to use a new style object in `styles.js. Thanks.